src/Components/OATGraphViewer/OATGraphViewer.tsx

```diff
@@ -108,6 +200,18 @@ const OATGraphViewer = ({ state, dispatch }: OATGraphProps) => {
         storeElements();
     }, [elements]);
 
+    // Update graph nodes and edges when the models are updated
+    useEffect(() => {
+        const potentialElements = getGraphViewerElementsFromModels(
+            models,
+            modelPositions
+        );
+
+        if (JSON.stringify(potentialElements) !== JSON.stringify(elements)) {
+            setElements(potentialElements);
+        }
+    }, [models]);
+
```

Comment:
Is there a way models can change outside this component as far as I can see we update Models on the translated output  and then translating them back into the element list right here but the translated output is a memo triggered by the element list